e2e-tests/test/anvil-apis.test.ts

```diff
@@ -0,0 +1,36 @@
+import { expect } from "chai";
+import { Wallet } from "zksync-web3";
+import { getTestProvider } from "../helpers/utils";
+import { ethers } from "hardhat";
+
+const provider = getTestProvider();
+
+describe("anvil_setBalance", function () {
```

Comment:
I think it worth adding tests for rest endpoints as well

Reply:
OK, that was a bit painful, since the hardhat tests weren't all that independent, but well, they should be... Note that `hardhat_reset`/`anvil_reset` still isn't tested - it doesn't fit into a parallel run, I don't know how to configure the testing framework to run it by itself, and I've been too lazy to implement it outside of it... :-)

Reply:
why do you think tests run in parallel?

Reply:
well, basically because adding a test for `hardhat_reset` didn't work - but I admit that could be caused by something else, too... Do you think e2e tests run serially?

Reply:
>Do you think e2e tests run serially?
yes

Reply:
>hardhat_reset didn't work

what did not work?

Reply:
I don't remember the exact problem, but adding 
```
describe("anvil_reset", function () {
  it("Should reset node", async function () {
    await provider.send("anvil_reset");
  });
});
```
to the testsuite results in the attached log.
[reset.log.zip](https://github.com/user-attachments/files/16623407/reset.log.zip)

Reply:
Please investigate why it fails and fix

Reply:
we have to have tests for all endpoints we add

Reply:
I'm afraid we aren't going to have automatic tests for `anvil_reset` - not this month, anyway... Do you want to remove it from the PR?

Reply:
I have opened PR that will cover this case https://github.com/matter-labs/era-test-node/pull/333

Reply:
> I have opened PR that will cover this case #333

Yes, but that PR has to be merged to use it here, right?